code/game/objects/structures/barricade/barricade.dm

```diff
@@ -459,30 +460,38 @@
 			repair_value = repair_materials[validSheetType]
 			break
 
+	if(!istype(material, /obj/item/stack/sheet/))
+		to_chat(user, SPAN_WARNING("You'll need some adequate repair material in your other hand to patch up [src]!"))
+		return FALSE
+
 	if(repair_value == 0)
 		to_chat(user, SPAN_WARNING("You'll need some adequate repair material in your other hand to patch up [src]!"))
 		return FALSE
 
-	var/soundchannel = playsound(src, nailgun.repair_sound, 25, 1)
-	if(!do_after(user, nailgun.nailing_speed, INTERRUPT_ALL, BUSY_ICON_FRIENDLY, src))
-		playsound(src, null, channel = soundchannel)
+	if(istype(item,/obj/item/weapon/gun/smg/nailgun/compact/tactical) && material.amount < mat_tac) //If less than [mat_tac] material is held, cancel repair.
+		to_chat(user, SPAN_WARNING("You need at least [mat_tac] [material.singular_name]\s to patch up [src]!"))
 		return FALSE
 
 	if(!material || (material != user.l_hand && material != user.r_hand) || material.amount <= 0)
-		to_chat(user, SPAN_WARNING("You seems to have misplaced the repair material!"))
+		to_chat(user, SPAN_WARNING("You seem to have misplaced [material]!"))
```

Comment:
Its not valid to try to insert `material` into a to_chat when this is a check if the `material` even exists.
```suggestion
		to_chat(user, SPAN_WARNING("You seem to have misplaced the repair material!"))
```